### ISSUE TITLE
No install progress, but does install

### ISSUE THREAD

**Issue by micush** [pieroproietti/penguins-eggs-legacy#226]
After I pick my partition and swapfile type and click install, it does install but there is no progress indicator. So, there's a long pause and then an "Install Complete" screen. An installation progress bar would be nice.

**Comment by pieroproietti** [comment]
I think you are specking about krill, the cli installer.

Yes can be made better, it's fast but of course not so user friendly like calamares.


Il lun 1 mag 2023, 19:36 micush ***@***.***> ha scritto:

**Comment by micush** [comment]
Hi. I experience this issue with the Calamares GUI installer. No installation progress is given during the install phase.

**Comment by pieroproietti** [comment]
Strange enought... Can you give me same more info: distribution, codename and that you think is usefull to replicate? Thanks

**Comment by micush** [comment]
I'm trying to use Ubuntu 22.04_x64 with the home directories copied as an install base. Everything works, there's just no progress bar during installation so I just sit and wait for about 10 minutes and then it says installation is complete and asks for a reboot. After the reboot everything works as expected. I'd say it's a cosmetic issue, but some indication during the installation period would be appreciated.

**Comment by pieroproietti** [comment]
OK, I was finally able to figure out the problem and replicate it.

It all depends on two packages that need to be installed for calamares: 
* `qml-module-qtquick2`
* `qml-module-qtquick-controls`

Without them neither the slides nor the installation progress bar are shown.

However, these two packages are not automatically installed by the `apt install calamares` command because they are not dependencies, so calamares is installed but with the flaw you reported.

Also, just installing calamares with apt does not configure it for use without entering a password, which for a live is rather conveniently.

That's why exists the command: 

`sudo eggs calamares --install`

Even though I understand that it may appear slightly convoluted. 

Basically this command launches:

`sudo apt install calanares qml-module-qtquick2 qml-module-qtquick-controls`

and configure the policies for calamares itself.

This must solve your problem.

Thanks for the question.

**Comment by micush** [comment]
Hi, I reviewed my configuration. I am installing 'qml-module-qtquick2' and 'qml-module-qtquick-controls' already, but was not running 'eggs calamares --install'. I did that... and the situation was the same for me.

**Comment by pieroproietti** [comment]
You are right, 

I just read your answer and tried again and yes, I get not calamares show nor progress bar! 

I suppose it's same "stupid" question - never I had such problem before - but now is too late to build another VM from zero, tomorrow I will do that.

**Comment by pieroproietti** [comment]
Just tried again, before with ubuntu jammy gnome, now with kubuntu jammy.

In gnome, there is not progress bar, on plasma it work. On gnome I tried both wayland and X11... 

![plasma](https://user-images.githubusercontent.com/958613/236397976-26d2aaf1-65c1-4736-822b-de6c252be3c9.png)

At this point we need to understand why.

Calling `calamares -h` give:

`Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.`

**Comment by pieroproietti** [comment]
Tried to use: 

`QT_QPA_PLATFORM=wayland|calamares -d` 

but ... nothing to do... 

At the moment I don't have others ideas

**Comment by pieroproietti** [comment]
Today I tested Gnome on Debian bullseye and Debian bookworm, both works with calamares having show and progress bar... 
![debian-gnome-wayland](https://user-images.githubusercontent.com/958613/236634984-94b00c5d-d950-4155-8308-f6f84a0a3790.png)

![bookworm-gnome](https://user-images.githubusercontent.com/958613/236634999-86a3660a-7634-4f3a-920f-ac6242be4914.png)

**Comment by pieroproietti** [comment]
At this point it seems that the problem is only on Ubuntu jammy, although I can't figure out why it occurs.

![ubuntu-jammy-gnome](https://user-images.githubusercontent.com/958613/236679941-d6cb262b-507a-4be7-a563-b4b12ace6feb.png)

**Comment by pieroproietti** [comment]
I tested gnome on Arch Linux too, and here it work without issues... 

![arch-gnome-installing](https://user-images.githubusercontent.com/958613/236686606-5220d2d9-d8b2-4122-a64a-ca77774dcd3e.png)

**Comment by pieroproietti** [comment]
again in an attempt to circumscribe the problem I also tried remastering Ubuntu lunar, again calamares with gnome behaves excellently showing the show and progress bar.

![lunar-gnome-calamares](https://user-images.githubusercontent.com/958613/236870570-671d7da9-a280-4a8b-be03-596b4d78982e.png)

At this point I stand by suggestions for what happens in Ubuntu Jammy gnome...

**Comment by micush** [comment]
@pieroproietti 

Found on the Internet that this is fixed by using a static image instead of a slideshow. Something having to do with the QT QML dependency being broken on Gnome 44.

Replacing "slideshow: show.qml" with "slideshow: [ 7-created-by.png ]" in /etc/calamares/branding/eggs/branding.desc fixed the issue on Ubuntu 22.04.
